### ISSUE TITLE
systemd-networkd: openvswitch issue

### ISSUE THREAD

**Issue by ne-vlezay80** [systemd/systemd#20241]
**systemd version the issue has been seen with**
 > systemd 249 (249-3-arch)

**Used distribution**
 > archlinux

**Linux kernel version used** (`uname -a`)

 > `5.12.15-arch1-1`

**CPU architecture issue was seen on**
 > amd64

**Expected behaviour you didn't see**
 > Inicialize network interface with systemd-networkd

**Unexpected behaviour you saw**
 > Interface not inicializing, if using systemd-networkd and openvswitrch

**Steps to reproduce the problem**
 > 1. Create  openvswitch bridge:
```
ovs-vsctl add-br lan
ovs-vsctl add-port lan net0
```
> 2. Create config from interface to add in openvswitch:
```
cat >>/etc/systemd/network/net0.network
[Match]
Name=net0

[Link]
MTUBytes=9000
```
> 3. systemctl restart systemd-networkd

**Additional program output to the terminal or log subsystem illustrating the issue**

```
Jul 18 13:59:15 router systemd-networkd[9434]: Could not drop routing policy rule: No such file or directory
Jul 18 13:59:15 router systemd-networkd[9434]: Could not drop routing policy rule: No such file or directory
Jul 18 13:59:15 router systemd-networkd[9434]: Could not drop routing policy rule: No such file or directory
Jul 18 13:59:15 router systemd-networkd[9434]: Could not drop routing policy rule: No such file or directory
Jul 18 13:59:15 router systemd-networkd[9434]: Could not drop routing policy rule: No such file or directory
Jul 18 13:59:15 router systemd-networkd[9434]: Could not drop routing policy rule: No such file or directory
Jul 18 13:59:15 router systemd-networkd[9434]: Could not drop routing policy rule: No such file or directory
Jul 18 13:59:15 router systemd-networkd[9434]: Could not drop routing policy rule: No such file or directory
Jul 18 13:59:15 router systemd-networkd[9434]: lan0: Failed to set master interface: Operation not supported
Jul 18 13:59:15 router systemd-networkd[9434]: lan0: Failed
Jul 18 13:59:15 router systemd-networkd[9434]: wan0: DHCPv4 address 100.64.9.95/21 via 100.64.8.1

```

**Comment by yuwata** [comment]
> Interface not inicializing, if using systemd-networkd and openvswitrch

Which interface is not initialized? `lan0`? `net0`?

Could you create the following drop-in config and provide the relevant debugging logs?
```
# /etc/systemd/system/systemd-networkd.service.d/override.conf
[Service]
Environment=SYSTEMD_LOG_LEVEL=debug
```
Also, please provide relevant .network file for `lan0`, if you are interested in `lan0`.

**Comment by ne-vlezay80** [comment]
> > Interface not inicializing, if using systemd-networkd and openvswitrch
> 
> Which interface is not initialized? `lan0`? `net0`?
> 
> Could you create the following drop-in config and provide the relevant debugging logs?
> 
> ```
> # /etc/systemd/system/systemd-networkd.service.d/override.conf
> [Service]
> Environment=SYSTEMD_LOG_LEVEL=debug
> ```
> 
> Also, please provide relevant .network file for `lan0`, if you are interested in `lan0`.

net0 is current interface name on configuration name.
lan0 equalent net0.

Because this log was removed before the configuration file was edited. 

There is no way to enable debugging, since this is the main server.

**Comment by ne-vlezay80** [comment]
Replayed from test lvirtual machine:
```
июл 19 07:46:10 testvm systemd-networkd[296]: lo: Link UP
июл 19 07:46:10 testvm systemd-networkd[296]: lo: Gained carrier
июл 19 07:46:10 testvm systemd-networkd[296]: null0: Gained IPv6LL
июл 19 07:46:10 testvm systemd-networkd[296]: ens3: Gained IPv6LL
июл 19 07:46:10 testvm systemd-networkd[296]: Enumeration completed
июл 19 07:46:10 testvm systemd[1]: Started Network Configuration.
июл 19 07:46:10 testvm systemd-networkd[296]: null0: netdev exists, using existing without changing its parameters
июл 19 07:46:10 testvm systemd-networkd[296]: null0: Failed to set master interface: Operation not supported
июл 19 07:46:10 testvm systemd-networkd[296]: null0: Failed
июл 19 07:46:11 testvm systemd-networkd[296]: ens3: DHCPv4 address 192.168.180.42/24 via 192.168.180.1
[root@testvm user]# ovs-vsctl show
ae700b32-5973-4777-a6d4-4e4086d189c7
    Bridge lan
        Port lan
            Interface lan
                type: internal
        Port null0
```

/etc/systemd/network/null0.netdev:
```
[root@testvm user]# cat /etc/systemd/network/null0.netdev 
[NetDev]
Name=null0
Kind=dummy
```

/etc/systemd/network/null0.network:
```
[Match]
Name=null0

[Link]
MTUBytes=16384

```

**Comment by blochl** [comment]
I can reproduce this issue.

My setup is (on Archlinux, kernel 5.13.5):
```
$ sudo ovs-vsctl show
7400290d-588e-4a24-9267-efde42fd4cb1
    Bridge br0
        Port eno1
            Interface eno1
        Port br0
            Interface br0
                type: internal
```
And the error:
```
Jul 29 10:25:35 lb-desktop systemd-networkd[438]: eno1: Failed to set master interface: Operation not supported
Jul 29 10:25:35 lb-desktop systemd-networkd[438]: eno1: Failed
```
It happens **most** of the time with systemd 249, and then I get no network. Never occurred on 248 and before.

**Comment by yuwata** [comment]
Fix is waiting in #20375.
